### PR TITLE
refactor(css/parser): refactor more

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Refactor parser more

**BREAKING CHANGE:**

No, but AST is changed

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/tests/recovery/hacks/output.swc-stderr [hunk 1]
```diff
@@ -149,48 +137,24 @@
...
-
   x Expected whitespace, semicolon, EOF, at-keyword or ident token
      ,-[$DIR/tests/recovery/hacks/input.css:226:1]
  226 | .selector { [;property: value;]; }
      :             ^
      `----
 
-  x Expected whitespace, semicolon, EOF, at-keyword or ident token
-     ,-[$DIR/tests/recovery/hacks/input.css:226:1]
- 226 | .selector { [;property: value;]; }
```
Comment (alexander-akait) [inline crates/swc_css_parser/tests/recovery/hacks/output.swc-stderr:190]: Avoid duplicate errors, we already reported about it above

**Comment by alexander-akait** [comment]
@kdy1 Ready for review

File: crates/swc_css_parser/tests/recovery/at-rule/font-face/span.rust-debug [hunk 2]
```diff
@@ -356,6 +356,18 @@
...
 
+  x ComponentValue
+    ,-[$DIR/tests/recovery/at-rule/font-face/input.css:17:1]
+ 17 | @font-face foo {}
+    :               ^
+    `----
+
+  x WhiteSpace { value: Atom(' ' type=inline) }
+    ,-[$DIR/tests/recovery/at-rule/font-face/input.css:17:1]
+ 17 | @font-face foo {}
```
Comment (kdy1) [inline crates/swc_css_parser/tests/recovery/at-rule/font-face/span.rust-debug:367]: I think this should be skipped

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_css_ast

**Pull request merged**